### PR TITLE
Optimize nonce validation in inbound channel

### PR DESCRIPTION
**Pull request by sumitsnk**
This PR resolves the issue SNO-147

### CODE REVIEW

File: ethereum/contracts/BasicInboundChannel.sol [hunk 1]
```diff
@@ -56,11 +56,14 @@ contract BasicInboundChannel {
     }
 
     function processMessages(Message[] calldata _messages) internal {
+        // Using tempNonce variable to store nonce value
```
Comment (vgeddes) [inline ethereum/contracts/BasicInboundChannel.sol:59]: This comment doesn't really explain anything. Instead, maybe explain that we are caching the nonce in a memory variable for gas optimization, or something like that.

File: ethereum/contracts/BasicInboundChannel.sol [hunk 2]
```diff
@@ -56,11 +56,14 @@ contract BasicInboundChannel {
...
+        uint64 tempNonce = nonce;
```
Comment (vgeddes) [inline ethereum/contracts/BasicInboundChannel.sol:60]: I would rather rename `tempNonce` to `cachedNonce`.

File: ethereum/contracts/IncentivizedInboundChannel.sol [hunk 3]
```diff
@@ -82,12 +82,13 @@ contract IncentivizedInboundChannel is AccessControl {
         Message[] calldata _messages
     ) internal {
         uint256 _rewardAmount = 0;
+        // Using tempNonce variable to store nonce value
+        uint64 tempNonce = nonce;
```
Comment (vgeddes) [inline ethereum/contracts/IncentivizedInboundChannel.sol:86]: See my comments about this same line in the basic channel.

**Review by vgeddes — COMMENTED** [review]
Looks good so far Sumit.

I think we need to add unit tests to guard against any future regressions in the cached nonce behavior. Could you update the following tests (or add new ones) to verify that the cached nonce was saved back to persistent storage.

https://github.com/Snowfork/snowbridge/blob/05fb58b7a2ed9d41884f2c4546cacdf87fb548c2/ethereum/test/test_basic_inbound_channel.js#L40

https://github.com/Snowfork/snowbridge/blob/05fb58b7a2ed9d41884f2c4546cacdf87fb548c2/ethereum/test/test_incentivized_inbound_channel.js#L47

**Comment by vgeddes** [comment]
Resolves SNO-147

File: ethereum/contracts/BasicInboundChannel.sol (hunk 1 above)
Comment (musnit) [reply ethereum/contracts/BasicInboundChannel.sol:59]: Yeh, in general comments should explain "Why" something is done, not "what" is being done. Variable names should explain "what" is being done.

**Review by musnit — APPROVED** [review]

**Review by vgeddes — APPROVED** [review]
Looks great! Go ahead and merge in.

**Pull request merged**